vividus-extension-selenium/src/main/java/org/vividus/steps/ui/BarcodeSteps.java

```diff
@@ -75,17 +91,66 @@ public BarcodeSteps(ScreenshotTaker screenshotTaker, BarcodeActions barcodeActio
+
+        scanScreenshot(viewportScreenshot, scopes, variableName);
+    }
+
+    /**
+     * Scan a barcode from context and save its value to the <b>variable</b> with the specified
+     * <b>variableName</b>
+     * Actions performed at this step:
+     * <ul>
+     * <li>Takes a screenshot the specified context. If it's not set - takes a screenshot of the entire page
+     * <li>Scans a barcode from the screenshot and save its value to the <i>variable name</i>
+     * </ul>
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario
+     * <li><b>STORY</b> - the variable will be available within the whole story
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name under which the value should be saved
+     * @throws IOException If an input or output exception occurred
+     */
+    @When("I scan barcode from context and save result to $scopes variable `$variableName`")
+    public void scanBarcodeFromContext(Set<VariableScope> scopes, String variableName) throws IOException
+    {
+        Optional<SearchContext> searchContext = uiContext.getOptionalSearchContext();
+
+        if (searchContext.isPresent())
+        {
+            Screenshot screenshot = ashotScreenshotTaker.takeAshotScreenshot(searchContext.get(), Optional.empty());
+            BufferedImage contextScreenshot = screenshot.getImage();
+
+            scanScreenshot(contextScreenshot, scopes, variableName);
+        }
+        else
+        {
+            LOGGER.warn("Could not find any context. The step will not be executed.");
+        }
+    }
+
+    private void scanScreenshot(BufferedImage screenshot, Set<VariableScope> scopes, String variableName)
+            throws IOException
+    {
         try
         {
-            String result = barcodeActions.scanBarcode(viewportScreenshot);
+            String result = barcodeActions.scanBarcode(screenshot);
             variableContext.putVariable(scopes, variableName, result);
         }
         catch (NotFoundException e)
         {
-            softAssert.recordFailedAssertion("There is no barcode on the screen", e);
-            Attachment attachment = new Attachment(ImageTool.toByteArray(viewportScreenshot), "Viewport Screenshot",
-                    "image/png");
-            eventBus.post(new AttachmentPublishEvent(attachment));
+            publishScreenshotWithUnsuitableContext(screenshot, e);
         }
     }
+
+    private void publishScreenshotWithUnsuitableContext(BufferedImage bufferedImage, NotFoundException exception)
+            throws IOException
+    {
+        softAssert.recordFailedAssertion("There is no barcode on the screen or context", exception);
```

Comment:
we can't easily change the assertion message, since users may rely on it in their known issues, so we need to keep the assertion message for the viewport-based step